login-workflow/src/new-architecture/screens/LoginScreen/LoginScreenBase.test.tsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+import { render, cleanup, screen, fireEvent } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import { LoginScreenBase } from './LoginScreenBase';
+
+const EMAIL_REGEX = /^[A-Z0-9._%+'-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i;
+
+afterEach(cleanup);
+
+const mockLogin = jest.fn();
+const mockForgotPassword = jest.fn();
+const mockSelfRegister = jest.fn();
+const mockContactSupport = jest.fn();
+const mockRememberMeChanged = jest.fn();
+
+describe('LoginScreenBase', () => {
+    beforeEach(() => {
+        render(
+            <LoginScreenBase
+                usernameLabel="Email Address"
+                usernameValidator={(username: string): string | boolean => {
+                    if (!EMAIL_REGEX.test(username)) {
+                        return 'Enter a valid email address';
+                    }
+                    return true;
+                }}
+                initialUsernameValue=""
+                passwordLabel="Password"
+                passwordValidator={(password: string): string | boolean => {
+                    if (password.length < 2) {
+                        return 'Password must be at least 2 characters';
+                    }
+                    return true;
+                }}
+                rememberMeLabel="Remember Me"
+                rememberMeInitialValue={true}
+                onRememberMeChanged={(value: boolean): void => {
+                    mockRememberMeChanged(value);
+                }}
+                loginButtonLabel="Log In"
+                onLogin={(username: string, password: string): void => {
+                    mockLogin(username, password);
+                }}
+                forgotPasswordLabel="Forgot your password?"
+                onForgotPassword={(): void => {
+                    mockForgotPassword();
+                }}
+                selfRegisterButtonLabel="Register now!"
+                selfRegisterInstructions="Need an account"
+                onSelfRegister={(): void => {
+                    mockSelfRegister();
+                }}
+                contactSupportLabel="Contact an Eaton Support Representative"
+                onContactSupport={(): void => {
+                    mockContactSupport();
+                }}
+            />
+        );
+    });
+
+    test('renders without crashing', () => {
+        expect(screen.getByText('Log In')).toBeInTheDocument();
+        expect(screen.getByText('Email Address')).toBeInTheDocument();
+        expect(screen.getByText('Password')).toBeInTheDocument();
+        // expect(screen.getByText('Forgot your password?')).toBeInTheDocument();
```

Comment:
not sure why the render isn't working here for Forgot your password. I tried doing this but no luck.
expect(screen.getByRole('link', { name: 'Forgot your password?' })).toBeInTheDocument();

Reply:
![image](https://github.com/etn-ccis/blui-react-workflows/assets/119693939/019ea5b0-840d-4af9-be9e-9100c2894e04)

could get closer to the mark if these could be included. not sure if its possible right now
![image](https://github.com/etn-ccis/blui-react-workflows/assets/119693939/81f8ed28-016e-4d07-b468-e72b8575d0f3)